org.torproject.torbrowser.yml

```diff
@@ -0,0 +1,50 @@
+app-id: org.torproject.torbrowser
+
+runtime: org.gnome.Platform
+runtime-version: "3.32"
+sdk: org.gnome.Sdk
+
+command: launch.sh
+
+finish-args:
+  - --socket=x11
+  # - --share=ipc
+  - --share=network
+
+modules:
+  - name: torbrowser
+    buildsystem: simple
+    build-commands:
+        - install -D Browser/browser/chrome/icons/default/default64.png   /app/share/icons/hicolor/64x64/apps/org.torproject.torbrowser.png
+        #- install -D Browser/start-tor-browser.desktop /app/share/applications/org.torproject.browser.desktop
+        - sed -i s,Icon=torbrowser,Icon=org.torproject.torbrowser,  torbrowser.desktop
+        - sed -i "s,Exec=torbrowser-launcher %u,Exec=/app/bin/launch.sh,"  torbrowser.desktop
+        - install -D torbrowser.appdata.xml  /app/share/metainfo/org.torproject.torbrowser.appdata.xml
+        - install -D torbrowser.desktop  /app/share/applications/org.torproject.torbrowser.desktop
+        - mv Browser /app
+        - install -D -m a=rx launch.sh   /app/bin/launch.sh
+    
+    cleanup:
+      - /Browser/TorBrowser/Docs/
+    sources:
+        - type: file
+          url: https://raw.githubusercontent.com/micahflee/torbrowser-launcher/develop/share/metainfo/torbrowser.appdata.xml
+          sha256: 4aa47248ce6ea3b54fe349bfa03169dc72cceb2f206d517cbf37eeab929d68d0
+
+        - type: file
+          url: https://raw.githubusercontent.com/micahflee/torbrowser-launcher/develop/share/applications/torbrowser.desktop
+          sha256: edc4a08adcfb78468a34bd46ef8f0de0ef5fdb141bd438709e755dc8b2a3eee6
+
+        - type: script
+          commands:
+            # Yes. This is dirty. The tarball contains a Firefox profile directory which Firefox expects to be able to write...
+            # The tmp directory is, at least when following this manifest and not overwriting the settings, a tmpfs, so we can (or have to) copy over everytime we launch.
+            - cp -ar --reflink=auto /app/Browser /tmp/
+            - cd /tmp/
+            - cd Browser
+            - ./start-tor-browser
+          dest-filename: launch.sh
+
+        - type: archive
+          url: https://www.torproject.org/dist/torbrowser/8.5.4/tor-browser-linux64-8.5.4_en-US.tar.xz
```

Comment:
Why not using [torbrowser-launcher](https://github.com/micahflee/torbrowser-launcher) instead, as mentioned before?

Reply:
what would be the benefits of doing so?

AFAIU, it's a wrapper around the real thing. A flatpak would be a wrapper of the wrapper. We could do that. But I'm not interested in creating a flatpak for said wrapper as I consider it to be quite pointless, because whatever is being shipped in `/app/` is not going to be run. Or rather, the wrapper performs the job that flatpak already performs, which is getting hold of the binary, verifying its integrity, and putting it into a well-known location (which, btw, the wrapper makes it in `/tmp/`, if `$HOME` does not exists, so you could indeed argue that `/tmp/`, and its volatile nature, is an okay thing to do).

That said, flatpaking the torbrowser-launcher should be simple. But flatpaking the launcher makes it a weird app that loads and executes another app. I had the feeling these kinds of applications are not very welcome on flathub. After all, you cannot run the Torbrowser after having installed the launcher through flatpak. You still need to run the launcher first and have it connect to the Internet.

What I think would be useful is to extract the version checking and downloading code in the torbrowser-launcher and make a manifest-generator out of that, s.t. running the generator creates a manifest with an up-to-date release.

Reply:
>flatpaking the launcher makes it a weird app that loads and executes another app

You are basically doing exactly the same, but worse: breaking UX, omitting languages, not getting always-latest version and not doing signature checks.